context.go

```diff
@@ -0,0 +1,231 @@
+package huma
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/danielgtaylor/huma/negotiation"
+	"github.com/fxamacker/cbor/v2"
+	"github.com/goccy/go-yaml"
+)
+
+// ContextFromRequest returns a Huma context for a request, useful for
+// accessing high-level convenience functions from e.g. middleware.
+func ContextFromRequest(w http.ResponseWriter, r *http.Request) Context {
+	return &hcontext{
+		Context:        r.Context(),
+		ResponseWriter: w,
+		r:              r,
+	}
+}
+
+// Context provides a request context and response writer with convenience
+// functions for error and model marshaling in handler functions.
+type Context interface {
+	context.Context
+	http.ResponseWriter
+
+	// AddError adds a new error to the list of errors for this request.
+	AddError(err error)
+
+	// HasError returns true if at least one error has been added to the context.
+	HasError() bool
+
+	// WriteError writes out an HTTP status code, friendly error message, and
+	// optionally a set of error details set with `AddError` and/or passed in.
+	WriteError(status int, message string, errors ...error)
+
+	// WriteModel writes out an HTTP status code and marshalled model based on
+	// content negotiation (e.g. JSON or CBOR). This must match the registered
+	// response status code & type.
+	WriteModel(status int, model interface{})
+}
+
+type hcontext struct {
+	context.Context
+	http.ResponseWriter
+	r      *http.Request
+	errors []error
+	op     *Operation
+	closed bool
+}
+
+func (c *hcontext) AddError(err error) {
+	c.errors = append(c.errors, err)
+}
+
+func (c *hcontext) HasError() bool {
+	return len(c.errors) > 0
+}
+
+func (c *hcontext) WriteHeader(status int) {
```

Comment:
Never understood why methods with this purpose are called "WriteHeader" in golang-world. You're not writing a "header" (as in a request or response header), you are writing a status code (or [status line](https://tools.ietf.org/html/rfc7230#section-3.1.2)). I find the name confusing.

Reply:
But I guess this implements `http.ResponseWriter` so 🤷

Reply:
Yeah this is purely to match up with the standard library `http.ResponseWriter` interface so we don't really control it :cry: